tools/docker/mu-plugins/allow_emails_from_localhost.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Plugin Name: Allow Emails from localhost
+ * Description: WordPress doesn't allow emails from localhost so we change the hostname when sending.
+ * Version: 1.0
+ * Author: Automattic
+ * Author URI: https://automattic.com/
+ * Text Domain: jetpack
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Jetpack\Docker\MuPlugin\AllowEmailsFromLocalhost;
+
+/**
+ * Allow sending emails from the Docker dev environment.
+ * WordPress relies on PHPMailer, which throws an exception when
+ * using localhost as the server name. This mocks a different
+ * email when needed.
+ *
+ * @param string $from_email Email address to send from.
+ *
+ * @return string Filtered email address to send from.
+ */
+function jetpack_allow_emails_from_localhost( $from_email ) {
+	if ( $from_email === 'wordpress@localhost' ) {
+		return 'wordpress@jetpack.docker';
```

Comment:
I wonder if we should change this to use a [reserved TLD](https://en.wikipedia.org/wiki/Top-level_domain#Reserved_domains) like `.localhost`, `.example`, `.test`, or `.internal`.

Looks like by default PHPMailer uses PHP's `filter_var()` with `FILTER_VALIDATE_EMAIL`, which is only rejecting `localhost` because it contains no dot. Even `wordpress@jetpack.localhost` should work.

Reply:
No objection here if you want to put up a PR.